### PR TITLE
AssemblyScript

### PR DESCRIPTION
**Pull request by trieloff**
try https://helix-git.edgecompute.app/?owner=adobe&repo=theblog

- [x] error handling
- [x] logging
- [x] proper format
- [x] CI/CD

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,3 @@
-coverage
-.nyc_output/
-node_modules/
-junit
-tmp
-logs
-.DS_Store
-test-results.xml
-dist
```
Comment (tripodsan) [inline .gitignore:9]: why did you remove this?

File: assembly/index.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return new Response(String.UTF8.encode("This method is not allowed"), {
+            status: 405,
+        });
+    }
+
+    let method = req.method();
+    let urlParts = req.url().split("//").pop().split("/");
+    let host = urlParts.shift();
+    let path = ("/" + urlParts.join("/"));
+    let qs = getQueryString(path);
```
Comment (tripodsan) [inline assembly/index.ts:51]: why not use a robust URL class to parse?

File: assembly/index.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    const owner = getQueryParam(qs, "owner");
+    const repo = getQueryParam(qs, "repo");
```
Comment (tripodsan) [inline assembly/index.ts:54]: why not use SearchParameters or querystring class?

File: assembly/index.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    let host = urlParts.shift();
+    let path = ("/" + urlParts.join("/"));
+    let qs = getQueryString(path);
+
+    const owner = getQueryParam(qs, "owner");
+    const repo = getQueryParam(qs, "repo"); 
+    let ref = getQueryParam(qs, "ref");
+    let sha = "";
+
+    if (owner != "" && repo != "" && true) {
```
Comment (tripodsan) [inline assembly/index.ts:58]: can we use the same code style rules also for typescript? eg single quotes etc.. ?

File: assembly/index.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          });
+    }
+
+    return new Response(String.UTF8.encode('Specify owner and repo, please.'), {
+        status: 400
+    });
+}
+
+// Get the request from the client.
+let req = Fastly.getClientRequest();
```
Comment (tripodsan) [inline assembly/index.ts:95]: `const`

File: assembly/index.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+// Pass the request to the main request handler function.
+let resp = main(req);
```
Comment (tripodsan) [inline assembly/index.ts:98]: `const`

**Comment by tripodsan** [comment]
I would move this to another repository.....

File: assembly/index.ts (hunk 2 above)
Comment (trieloff) [reply assembly/index.ts:51]: There isn't one in the AssemblyScript standard library: https://www.assemblyscript.org/stdlib/globals.html

File: assembly/index.ts (hunk 3 above)
Comment (trieloff) [reply assembly/index.ts:54]: There isn't one in the AssemblyScript standard library: https://www.assemblyscript.org/stdlib/globals.html

**Comment by stefan-guggisberg** [comment]
Why a destructive PR which replaces the current OW-based implementation?

**Comment by trieloff** [comment]
> Why a destructive PR which replaces the current OW-based implementation?

Because I haven't found a good way to have both in the same repo. We probably should keep the branch, but close the PR if we want to keep both side by side

**Comment by stefan-guggisberg** [comment]
@trieloff are you still working on this?

**Comment by trieloff** [comment]
We probably won't use it anytime soon, so I'll close the PR, but keep the branch.
